autoload/go/lint.vim

```diff
@@ -44,15 +36,17 @@ function! go#lint#Gometa(bang, autosave, ...) abort
     " will be cleared
     redraw
 
-    " Include only messages for the active buffer for autosave.
-    let include = [printf('--include=^%s:.*$', fnamemodify(expand('%:p'), ":."))]
-    if go#util#has_job()
-      let include = [printf('--include=^%s:.*$', expand('%:p:t'))]
+    if l:metalinter == "gometalinter"
+      " Include only messages for the active buffer for autosave.
+      let include = [printf('--include=^%s:.*$', fnamemodify(expand('%:p'), ":."))]
+      if go#util#has_job()
+        let include = [printf('--include=^%s:.*$', expand('%:p:t'))]
+      endif
+      let cmd += include
```

Comment:
How can we ensure that only messages for the active buffer are included when using golangci-lint?

Reply:
they unfortunately don't have an `include` like gometalinter. i'll take another look at their flag options to see if there's something we can do

Reply:
figured this out, golangci-lint can be run per file so I added https://github.com/fatih/vim-go/pull/2182/commits/f574b1f06dee7a23483e79f55c956c83763ba866